core/src/test/java/com/linecorp/armeria/common/encoding/StreamDecoderCompatibleTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.encoding;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.encoding.DecodingClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.encoding.EncodingService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+class StreamDecoderCompatibleTest {
```

Comment:
How about renaming this class into `DecodingClientTest` and adding the tests for both deprecated and non-deprecated stream decoders? I'm just not sure why we don't have `DecodingClientTest` already. 😅

Reply:
renaming and add some test (https://github.com/line/armeria/pull/3087/commits/d1eb993f9b129d112a124dff02948c6b558f91dc)
add compatible test for `HttpDecodedResponse` (https://github.com/line/armeria/pull/3087/commits/f7d9b4457a1469471b0999bbc11bbe7d665845bf)